commit 48527cdf1b9110c89e8f483c2b8f34dfa058b6ff
Author: yang <yang@fydeos.io>
Date:   Thu Jul 11 16:56:14 2024 +0800

    fix the issue kernel can't init panel if uboot display some bmp; add new mode: pressing volume-down enter ums mode;

diff --git a/arch/arm/dts/rk3588s-fydetab-duo.dts b/arch/arm/dts/rk3588s-fydetab-duo.dts
index ff1a4eec3f..aa028d261d 100644
--- a/arch/arm/dts/rk3588s-fydetab-duo.dts
+++ b/arch/arm/dts/rk3588s-fydetab-duo.dts
@@ -40,6 +40,13 @@
 			label = "volume up";
 			press-threshold-microvolt = <1750>;
 		};
+
+    volumedown-key {
+      u-boot,dm-pre-reloc;
+      linux,code = <KEY_VOLUMEDOWN>;
+      label = "volume down";
+      press-threshold-microvolt = <500000>;
+    };
 	};
 
 	vcc_5v0: vcc-5v0 {
@@ -64,7 +71,7 @@
 		pinctrl-0 = <&vcc_lcd_en_pin>;
 		vin-supply = <&vcc_5v0>;
 	};
-	
+
 	vcc_sd: vcc-sd-power {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_sd";
@@ -136,7 +143,7 @@
 		#phy-cells = <0>;
 		status = "okay";
 	};
-	
+
 	vop: vop@fdd90000 {
 		compatible = "rockchip,rk3588-vop";
 		reg = <0x0 0xfdd90000 0x0 0x4200>, <0x0 0xfdd95000 0x0 0x1000>;
@@ -234,8 +241,8 @@
 
 			vp2: port@2 {
 				cursor-win-id=<ROCKCHIP_VOP2_ESMART2>;
-				rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER2 | 1 << ROCKCHIP_VOP2_ESMART2)>;
-				rockchip,primary-plane = <ROCKCHIP_VOP2_CLUSTER2>;
+				rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER3 | 1 << ROCKCHIP_VOP2_ESMART2)>;
+				rockchip,primary-plane = <ROCKCHIP_VOP2_CLUSTER3>;
 				#address-cells = <1>;
 				#size-cells = <0>;
 				reg = <2>;
@@ -267,8 +274,8 @@
 
 			vp3: port@3 {
 				cursor-win-id=<ROCKCHIP_VOP2_ESMART3>;
-				rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER3 | 1 << ROCKCHIP_VOP2_ESMART3)>;
-				rockchip,primary-plane = <ROCKCHIP_VOP2_CLUSTER3>;
+				rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER2 | 1 << ROCKCHIP_VOP2_ESMART3)>;
+				rockchip,primary-plane = <ROCKCHIP_VOP2_CLUSTER2>;
 				#address-cells = <1>;
 				#size-cells = <0>;
 				reg = <3>;
@@ -346,10 +353,10 @@
 
 			route_dsi0: route-dsi0 {
 				status = "okay";
-				logo,uboot = "logo.bmp";
-				logo,kernel = "logo_kernel.bmp";
-				logo,mode = "center";
-				charge_logo,mode = "center";
+				logo,uboot = "fydetab_recovery.bmp";
+				logo,kernel = "fydetab_usb.bmp";
+				logo,mode = "fullscreen";
+				charge_logo,mode = "fullscreen";
 				connect = <&vp2_out_dsi0>;
 			};
 
@@ -416,14 +423,14 @@
 		reg = <0>;
 
 		power-supply = <&vcc_lcd_en>;
-		reset-gpios = <&gpio3 RK_PC6 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
 		avdd-gpios = <&gpio3 RK_PA7 GPIO_ACTIVE_HIGH>;
 		avee-gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&avdd_lcd_gpio &avee_lcd_gpio &lcd_rst_gpio>;
-		
-		prepare-delay-ms = <2>;
-		reset-delay-ms = <2>;
+
+		prepare-delay-ms = <20>;
+		reset-delay-ms = <12>;
 		init-delay-ms = <65>;
 		enable-delay-ms = <40>;
 
@@ -623,4 +630,4 @@
 				<1 RK_PA6 5 &pcfg_pull_none>;
 		};
 	};
-};
\ No newline at end of file
+};
diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index 6121bb8ce5..1b08029249 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -473,9 +473,11 @@ int board_late_init(void)
 #endif
 
 #ifdef CONFIG_DRM_ROCKCHIP
+#if 0
 	if (rockchip_get_boot_mode() != BOOT_MODE_QUIESCENT)
 		rockchip_show_logo();
 #endif
+#endif
 #ifdef CONFIG_ROCKCHIP_EINK_DISPLAY
 	rockchip_eink_show_uboot_logo();
 #endif
@@ -628,6 +630,9 @@ void arch_preboot_os(uint32_t bootm_state, bootm_headers_t *images)
 	if (!(bootm_state & BOOTM_STATE_OS_PREP))
 		return;
 
+#ifdef CONFIG_DRM_ROCKCHIP
+  rockchip_close_display();
+#endif
 #ifdef CONFIG_ARM64
 	u8 *data = (void *)images->ep;
 
diff --git a/arch/arm/mach-rockchip/boot_rkimg.c b/arch/arm/mach-rockchip/boot_rkimg.c
index 40c91a6ac3..8a82a29777 100644
--- a/arch/arm/mach-rockchip/boot_rkimg.c
+++ b/arch/arm/mach-rockchip/boot_rkimg.c
@@ -295,7 +295,7 @@ void rockchip_set_bootdev(struct blk_desc *desc)
  * some special boards.
  */
 #define KEY_DOWN_MIN_VAL	0
-#define KEY_DOWN_MAX_VAL	30
+#define KEY_DOWN_MAX_VAL	20000
 
 __weak int rockchip_dnl_key_pressed(void)
 {
@@ -325,7 +325,33 @@ __weak int rockchip_dnl_key_pressed(void)
 	return 0;
 }
 
-int rockchip_show_loader_logo(void);
+__weak int rockchip_ums_key_pressed(void)
+{
+#if defined(CONFIG_DM_KEY)
+  return key_is_pressed(key_read(KEY_VOLUMEDOWN));
+
+#elif defined(CONFIG_ADC)
+  const void *blob = gd->fdt_blob;
+  int node, ret, channel = 1;
+  u32 val, chns[2];
+
+  node = fdt_node_offset_by_compatible(blob, 0, "adc-keys");
+  if (node >= 0) {
+    if (!fdtdec_get_int_array(blob, node, "io-channels", chns, 2))
+      channel = chns[1];
+  }
+
+  ret = adc_channel_single_shot("saradc", channel, &val);
+  if (ret) {
+    printf("%s: Failed to read saradc, ret=%d\n", __func__, ret);
+    return 0;
+  }
+
+  return ((val >= 20000) && (val <= 500000));
+#endif
+
+  return 0;
+}
 
 void setup_download_mode(void)
 {
@@ -343,7 +369,7 @@ void setup_download_mode(void)
 			printf("%sentering download mode...\n",
 			       IS_ENABLED(CONFIG_CMD_ROCKUSB) ?
 			       "" : "no rockusb, ");
-			rockchip_show_loader_logo();
+			rockchip_show_logo();
 
 			/* try rockusb download and brom download */
 			run_command("download", 0);
@@ -351,6 +377,9 @@ void setup_download_mode(void)
 			printf("entering recovery mode!\n");
 			env_set("reboot_mode", "recovery-key");
 		}
+	} else if (rockchip_ums_key_pressed()) {
+		rockchip_show_bmp("fydetab_usb.bmp");
+		run_command("ums 0 mmc 0", 0);
 	} else if (is_hotkey(HK_FASTBOOT)) {
 		env_set("reboot_mode", "fastboot");
 	}
diff --git a/configs/rk3588s_fydetab_duo_defconfig b/configs/rk3588s_fydetab_duo_defconfig
index bf8709f5a1..f047365eb1 100644
--- a/configs/rk3588s_fydetab_duo_defconfig
+++ b/configs/rk3588s_fydetab_duo_defconfig
@@ -1,243 +1,1584 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# U-Boot 2017.09 Configuration
+#
+CONFIG_CREATE_ARCH_SYMLINK=y
+# CONFIG_ARC is not set
 CONFIG_ARM=y
+# CONFIG_M68K is not set
+# CONFIG_MICROBLAZE is not set
+# CONFIG_MIPS is not set
+# CONFIG_NDS32 is not set
+# CONFIG_NIOS2 is not set
+# CONFIG_PPC is not set
+# CONFIG_RISCV is not set
+# CONFIG_SANDBOX is not set
+# CONFIG_SH is not set
+# CONFIG_X86 is not set
+# CONFIG_XTENSA is not set
+CONFIG_SYS_ARCH="arm"
+CONFIG_SYS_CPU="armv8"
+CONFIG_SYS_SOC="rockchip"
+CONFIG_SYS_VENDOR="rockchip"
+CONFIG_SYS_BOARD="evb_rk3588"
+CONFIG_SYS_CONFIG_NAME="evb_rk3588"
+
+#
+# ARM architecture
+#
+CONFIG_ARM64=y
+# CONFIG_POSITION_INDEPENDENT is not set
+CONFIG_STATIC_RELA=y
+CONFIG_DMA_ADDR_T_64BIT=y
+CONFIG_ARM_ASM_UNIFIED=y
+CONFIG_SYS_ARM_ARCH=8
+CONFIG_SYS_CACHE_SHIFT_6=y
+CONFIG_SYS_CACHELINE_SIZE=64
+CONFIG_ARM_SMCCC=y
+CONFIG_ARM_CPU_SUSPEND=y
+# CONFIG_SEMIHOSTING is not set
+# CONFIG_SYS_L2CACHE_OFF is not set
+CONFIG_ENABLE_ARM_SOC_BOOT0_HOOK=y
+# CONFIG_ARM_CORTEX_CPU_IS_UP is not set
+CONFIG_ARM64_SUPPORT_AARCH32=y
+# CONFIG_ARM_SMP is not set
+# CONFIG_ARCH_AT91 is not set
+# CONFIG_TARGET_EDB93XX is not set
+# CONFIG_TARGET_ASPENITE is not set
+# CONFIG_TARGET_GPLUGD is not set
+# CONFIG_ARCH_DAVINCI is not set
+# CONFIG_KIRKWOOD is not set
+# CONFIG_ARCH_MVEBU is not set
+# CONFIG_TARGET_DEVKIT3250 is not set
+# CONFIG_TARGET_WORK_92105 is not set
+# CONFIG_TARGET_MX25PDK is not set
+# CONFIG_TARGET_ZMX25 is not set
+# CONFIG_TARGET_APF27 is not set
+# CONFIG_TARGET_APX4DEVKIT is not set
+# CONFIG_TARGET_XFI3 is not set
+# CONFIG_TARGET_M28EVK is not set
+# CONFIG_TARGET_MX23EVK is not set
+# CONFIG_TARGET_MX28EVK is not set
+# CONFIG_TARGET_MX23_OLINUXINO is not set
+# CONFIG_TARGET_BG0900 is not set
+# CONFIG_TARGET_SANSA_FUZE_PLUS is not set
+# CONFIG_TARGET_SC_SPS_1 is not set
+# CONFIG_ORION5X is not set
+# CONFIG_TARGET_SPEAR300 is not set
+# CONFIG_TARGET_SPEAR310 is not set
+# CONFIG_TARGET_SPEAR320 is not set
+# CONFIG_TARGET_SPEAR600 is not set
+# CONFIG_TARGET_STV0991 is not set
+# CONFIG_TARGET_X600 is not set
+# CONFIG_TARGET_IMX31_PHYCORE is not set
+# CONFIG_TARGET_IMX31_PHYCORE_EET is not set
+# CONFIG_TARGET_MX31ADS is not set
+# CONFIG_TARGET_MX31PDK is not set
+# CONFIG_TARGET_WOODBURN is not set
+# CONFIG_TARGET_WOODBURN_SD is not set
+# CONFIG_TARGET_FLEA3 is not set
+# CONFIG_TARGET_MX35PDK is not set
+# CONFIG_ARCH_BCM283X is not set
+# CONFIG_TARGET_VEXPRESS_CA15_TC2 is not set
+# CONFIG_ARCH_BCMSTB is not set
+# CONFIG_TARGET_VEXPRESS_CA5X2 is not set
+# CONFIG_TARGET_VEXPRESS_CA9X4 is not set
+# CONFIG_TARGET_BCM23550_W1D is not set
+# CONFIG_TARGET_BCM28155_AP is not set
+# CONFIG_TARGET_BCMCYGNUS is not set
+# CONFIG_TARGET_BCMNSP is not set
+# CONFIG_TARGET_BCMNS2 is not set
+# CONFIG_ARCH_EXYNOS is not set
+# CONFIG_ARCH_S5PC1XX is not set
+# CONFIG_ARCH_HIGHBANK is not set
+# CONFIG_ARCH_INTEGRATOR is not set
+# CONFIG_ARCH_KEYSTONE is not set
+# CONFIG_ARCH_OMAP2PLUS is not set
+# CONFIG_ARCH_MESON is not set
+# CONFIG_ARCH_MX7ULP is not set
+# CONFIG_ARCH_MX7 is not set
+# CONFIG_ARCH_MX6 is not set
+CONFIG_SPL_LDSCRIPT="arch/arm/cpu/armv8/u-boot-spl.lds"
+# CONFIG_ARCH_MX5 is not set
+# CONFIG_ARCH_RMOBILE is not set
+# CONFIG_TARGET_S32V234EVB is not set
+# CONFIG_ARCH_SNAPDRAGON is not set
+# CONFIG_ARCH_SOCFPGA is not set
+# CONFIG_ARCH_SUNXI is not set
+# CONFIG_TARGET_TS4600 is not set
+# CONFIG_ARCH_VF610 is not set
+# CONFIG_ARCH_ZYNQ is not set
+# CONFIG_ARCH_ZYNQMP is not set
+# CONFIG_TEGRA is not set
+# CONFIG_TARGET_VEXPRESS64_AEMV8A is not set
+# CONFIG_TARGET_VEXPRESS64_BASE_FVP is not set
+# CONFIG_TARGET_VEXPRESS64_BASE_FVP_DRAM is not set
+# CONFIG_TARGET_VEXPRESS64_JUNO is not set
+# CONFIG_TARGET_LS2080A_EMU is not set
+# CONFIG_TARGET_LS2080A_SIMU is not set
+# CONFIG_TARGET_LS2080AQDS is not set
+# CONFIG_TARGET_LS2080ARDB is not set
+# CONFIG_TARGET_LS2081ARDB is not set
+# CONFIG_TARGET_HIKEY is not set
+# CONFIG_TARGET_POPLAR is not set
+# CONFIG_TARGET_LS1012AQDS is not set
+# CONFIG_TARGET_LS1012ARDB is not set
+# CONFIG_TARGET_LS1012AFRDM is not set
+# CONFIG_TARGET_LS1021AQDS is not set
+# CONFIG_TARGET_LS1021ATWR is not set
+# CONFIG_TARGET_LS1021AIOT is not set
+# CONFIG_TARGET_LS1043AQDS is not set
+# CONFIG_TARGET_LS1043ARDB is not set
+# CONFIG_TARGET_LS1046AQDS is not set
+# CONFIG_TARGET_LS1046ARDB is not set
+# CONFIG_TARGET_H2200 is not set
+# CONFIG_TARGET_ZIPITZ2 is not set
+# CONFIG_TARGET_COLIBRI_PXA270 is not set
+# CONFIG_ARCH_UNIPHIER is not set
+# CONFIG_STM32 is not set
+# CONFIG_ARCH_STI is not set
 CONFIG_ARCH_ROCKCHIP=y
+# CONFIG_TARGET_THUNDERX_88XX is not set
+# CONFIG_ARCH_ASPEED is not set
 CONFIG_SPL_GPIO_SUPPORT=y
 CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
 CONFIG_SYS_MALLOC_F_LEN=0x80000
+# CONFIG_ROCKCHIP_PX30 is not set
+CONFIG_TPL_LDSCRIPT="arch/arm/mach-rockchip/u-boot-tpl-v8.lds"
+CONFIG_TPL_TEXT_BASE=0xff001000
+CONFIG_TPL_MAX_SIZE=61440
+# CONFIG_ROCKCHIP_RK3036 is not set
+# CONFIG_ROCKCHIP_RK3128 is not set
+# CONFIG_ROCKCHIP_RK3066 is not set
+# CONFIG_ROCKCHIP_RK3188 is not set
+# CONFIG_ROCKCHIP_RK322X is not set
+# CONFIG_ROCKCHIP_RK3288 is not set
 CONFIG_SPL_FIT_GENERATOR="arch/arm/mach-rockchip/make_fit_atf.sh"
+# CONFIG_ROCKCHIP_RK3308 is not set
+# CONFIG_ROCKCHIP_RK3328 is not set
+# CONFIG_ROCKCHIP_RK3368 is not set
+# CONFIG_ROCKCHIP_RK3399 is not set
+# CONFIG_ROCKCHIP_RK3528 is not set
+# CONFIG_ROCKCHIP_RK3562 is not set
+# CONFIG_ROCKCHIP_RK3568 is not set
 CONFIG_ROCKCHIP_RK3588=y
+# CONFIG_ROCKCHIP_RK1808 is not set
+# CONFIG_ROCKCHIP_RV1106 is not set
+# CONFIG_ROCKCHIP_RV1108 is not set
+# CONFIG_ROCKCHIP_RV1126 is not set
+# CONFIG_SPL_ROCKCHIP_BACK_TO_BROM is not set
+CONFIG_TPL_ROCKCHIP_BACK_TO_BROM=y
+# CONFIG_ARM64_BOOT_AARCH32 is not set
+# CONFIG_ARM64_SWITCH_TO_AARCH32 is not set
+CONFIG_ROCKCHIP_BOOT_MODE_REG=0xfd588080
+CONFIG_ROCKCHIP_STIMER_BASE=0xfd8c8000
+CONFIG_ROCKCHIP_IRAM_START_ADDR=0xff000000
+CONFIG_ROCKCHIP_SPL_RESERVE_IRAM=0
+CONFIG_ROCKCHIP_BROM_HELPER=y
+# CONFIG_SPL_ROCKCHIP_EARLYRETURN_TO_BROM is not set
+# CONFIG_TPL_ROCKCHIP_EARLYRETURN_TO_BROM is not set
+CONFIG_SPL_MMC_SUPPORT=y
+CONFIG_RKIMG_BOOTLOADER=y
+# CONFIG_RKIMG_ANDROID_BOOTMODE_LEGACY is not set
+CONFIG_ROCKCHIP_BOOTDEV=""
+CONFIG_ROCKCHIP_RESOURCE_IMAGE=y
+CONFIG_ROCKCHIP_DTB_VERIFY=y
+# CONFIG_ROCKCHIP_USB_BOOT is not set
 CONFIG_ROCKCHIP_FIT_IMAGE=y
+# CONFIG_ROCKCHIP_UIMAGE is not set
+# CONFIG_ROCKCHIP_EARLY_DISTRO_DTB is not set
 CONFIG_ROCKCHIP_HWID_DTB=y
 CONFIG_ROCKCHIP_VENDOR_PARTITION=y
+CONFIG_USING_KERNEL_DTB=y
 CONFIG_USING_KERNEL_DTB_V2=y
 CONFIG_EMBED_KERNEL_DTB=y
 CONFIG_EMBED_KERNEL_DTB_PATH="arch/arm/dts/rk3588s-fydetab-duo.dtb"
 CONFIG_EMBED_KERNEL_DTB_ALWAYS=y
+# CONFIG_ROCKCHIP_CRC is not set
+CONFIG_ROCKCHIP_SMCCC=y
+# CONFIG_ROCKCHIP_DEBUGGER is not set
+# CONFIG_ROCKCHIP_CRASH_DUMP is not set
+CONFIG_ROCKCHIP_PRELOADER_ATAGS=y
+CONFIG_ROCKCHIP_PRELOADER_SERIAL=y
+# CONFIG_ROCKCHIP_IMAGE_TINY is not set
 CONFIG_ROCKCHIP_FIT_IMAGE_PACK=y
+CONFIG_ROCKCHIP_UART_MUX_SEL_M=0
+# CONFIG_ROCKCHIP_REBOOT_TEST is not set
 CONFIG_ROCKCHIP_NEW_IDB=y
+CONFIG_ROCKCHIP_HOTKEY=y
+CONFIG_ROCKCHIP_CMD=""
+# CONFIG_GICV2 is not set
+CONFIG_GICV3=y
+# CONFIG_ROCKCHIP_EMMC_IOMUX is not set
+# CONFIG_ROCKCHIP_NAND_IOMUX is not set
+# CONFIG_ROCKCHIP_SFC_IOMUX is not set
+CONFIG_ROCKCHIP_SET_SN=y
+CONFIG_ROCKCHIP_SET_ETHADDR=y
+CONFIG_BASE_DEFCONFIG=""
+CONFIG_CHIP_NAME=""
+CONFIG_LOADER_INI=""
+CONFIG_TRUST_INI=""
 CONFIG_PSTORE=y
+CONFIG_PERSISTENT_RAM_ADDR=0x0
+# CONFIG_ROCKCHIP_MINIDUMP is not set
 CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_BOARD_SPECIFIC_OPTIONS=y
+# CONFIG_TPL_LIBCOMMON_SUPPORT is not set
+# CONFIG_TPL_LIBGENERIC_SUPPORT is not set
 CONFIG_SPL_DRIVERS_MISC_SUPPORT=y
+CONFIG_ENV_SIZE=0x8000
 CONFIG_TARGET_EVB_RK3588=y
 CONFIG_SPL_LIBDISK_SUPPORT=y
+# CONFIG_SPL_NAND_SUPPORT is not set
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
+# CONFIG_SPL_WATCHDOG_SUPPORT is not set
+# CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_PARTITION_TYPE is not set
+CONFIG_IDENT_STRING=""
+# CONFIG_SPL_FAT_SUPPORT is not set
+# CONFIG_ARMV8_MULTIENTRY is not set
+# CONFIG_ARMV8_SET_SMPEN is not set
+
+#
+# ARMv8 secure monitor firmware
+#
+# CONFIG_ARMV8_SEC_FIRMWARE_SUPPORT is not set
+# CONFIG_SPL_ARMV8_SEC_FIRMWARE_SUPPORT is not set
+CONFIG_PSCI_RESET=y
+# CONFIG_ARMV8_PSCI is not set
+# CONFIG_ARMV8_CRYPTO is not set
+# CONFIG_CMD_DEKBLOB is not set
+# CONFIG_CMD_HDMIDETECT is not set
+
+#
+# ARM debug
+#
 CONFIG_DEFAULT_DEVICE_TREE="rk3588s-fydetab-duo"
 CONFIG_DEBUG_UART=y
+# CONFIG_AHCI is not set
+
+#
+# General setup
+#
+CONFIG_LOCALVERSION=""
+CONFIG_LOCALVERSION_AUTO=y
+CONFIG_CC_OPTIMIZE_FOR_SIZE=y
+CONFIG_DISTRO_DEFAULTS=y
+CONFIG_SYS_MALLOC_F=y
+CONFIG_SPL_SYS_MALLOC_F_LEN=0x80000
+CONFIG_TPL_SYS_MALLOC_F_LEN=0x80000
+CONFIG_EXPERT=y
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+# CONFIG_TOOLS_DEBUG is not set
+CONFIG_PHYS_64BIT=y
+
+#
+# Boot images
+#
+# CONFIG_IMAGE_GZIP is not set
+CONFIG_ANDROID_BOOT_IMAGE=y
 CONFIG_FIT=y
+CONFIG_FIT_ENABLE_SHA256_SUPPORT=y
+# CONFIG_FIT_ENABLE_RSA4096_SUPPORT is not set
+# CONFIG_FIT_SIGNATURE is not set
+# CONFIG_FIT_VERBOSE is not set
+# CONFIG_FIT_BEST_MATCH is not set
 CONFIG_FIT_IMAGE_POST_PROCESS=y
 CONFIG_FIT_HW_CRYPTO=y
+CONFIG_FIT_PRINT=y
+# CONFIG_FIT_OMIT_UBOOT is not set
+CONFIG_SPL_FIT=y
+# CONFIG_SPL_FIT_SIGNATURE is not set
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_FIT_IMAGE_POST_PROCESS=y
 CONFIG_SPL_FIT_HW_CRYPTO=y
 # CONFIG_SPL_SYS_DCACHE_OFF is not set
-CONFIG_BOOTDELAY=0
+# CONFIG_SPL_FIT_PRINT is not set
+CONFIG_SPL_FIT_SOURCE=""
+CONFIG_SPL_FIT_IMAGE_KB=2048
+CONFIG_SPL_FIT_IMAGE_MULTIPLE=2
+# CONFIG_OF_BOARD_SETUP is not set
+# CONFIG_OF_SYSTEM_SETUP is not set
+# CONFIG_OF_STDOUT_VIA_ALIAS is not set
+CONFIG_SYS_EXTRA_OPTIONS=""
+CONFIG_ARCH_FIXUP_FDT_MEMORY=y
+
+#
+# API
+#
+# CONFIG_API is not set
+
+#
+# Boot timing
+#
+# CONFIG_BOOTSTAGE is not set
+CONFIG_BOOTSTAGE_USER_COUNT=20
+CONFIG_BOOTSTAGE_RECORD_COUNT=30
+CONFIG_BOOTSTAGE_STASH_ADDR=0
+CONFIG_BOOTSTAGE_STASH_SIZE=0x1000
+# CONFIG_BOOTSTAGE_PRINTF_TIMESTAMP is not set
+
+#
+# Boot media
+#
+# CONFIG_NAND_BOOT is not set
+# CONFIG_ONENAND_BOOT is not set
+# CONFIG_QSPI_BOOT is not set
+# CONFIG_SATA_BOOT is not set
+# CONFIG_SD_BOOT is not set
+# CONFIG_SPI_BOOT is not set
+CONFIG_PASS_DEVICE_SERIAL_BY_FDT=y
+CONFIG_BOOTDELAY=1
+# CONFIG_USE_BOOTARGS is not set
+# CONFIG_IO_TRACE is not set
+
+#
+# Console
+#
+# CONFIG_CONSOLE_RECORD is not set
+# CONFIG_CONSOLE_DISABLE_CLI is not set
+# CONFIG_DISABLE_CONSOLE is not set
+CONFIG_LOGLEVEL=4
+CONFIG_SPL_LOGLEVEL=4
+# CONFIG_SILENT_CONSOLE is not set
+# CONFIG_PRE_CONSOLE_BUFFER is not set
+CONFIG_CONSOLE_MUX=y
+CONFIG_SYS_CONSOLE_IS_IN_ENV=y
+# CONFIG_SYS_CONSOLE_OVERWRITE_ROUTINE is not set
+# CONFIG_SYS_CONSOLE_ENV_OVERWRITE is not set
 CONFIG_SYS_CONSOLE_INFO_QUIET=y
+# CONFIG_SYS_STDIO_DEREGISTER is not set
+
+#
+# Logging
+#
+# CONFIG_LOG is not set
+# CONFIG_SPL_LOG is not set
+CONFIG_DEFAULT_FDT_FILE=""
+# CONFIG_VERSION_VARIABLE is not set
+CONFIG_BOARD_LATE_INIT=y
 # CONFIG_DISPLAY_CPUINFO is not set
+CONFIG_DISPLAY_BOARDINFO=y
+
+#
+# Start-up hooks
+#
+# CONFIG_ARCH_EARLY_INIT_R is not set
+# CONFIG_ARCH_MISC_INIT is not set
+# CONFIG_BOARD_EARLY_INIT_F is not set
+# CONFIG_BOARD_EARLY_INIT_R is not set
+
+#
+# Android features support
+#
 CONFIG_ANDROID_BOOTLOADER=y
+# CONFIG_ANDROID_AB is not set
+CONFIG_ANDROID_WRITE_KEYBOX=y
 CONFIG_ANDROID_AVB=y
+# CONFIG_ANDROID_AVB_ROLLBACK_INDEX is not set
+CONFIG_ANDROID_KEYMASTER_CA=y
 CONFIG_ANDROID_BOOT_IMAGE_HASH=y
+# CONFIG_SKIP_RELOCATE_UBOOT is not set
+
+#
+# Security support
+#
+CONFIG_HASH=y
+# CONFIG_BOARD_RNG_SEED is not set
+
+#
+# MT support
+#
+# CONFIG_MP_BOOT is not set
+
+#
+# SPL / TPL
+#
+CONFIG_SUPPORT_SPL=y
+CONFIG_SUPPORT_TPL=y
+CONFIG_SPL=y
+# CONFIG_SPL_ADC_SUPPORT is not set
+# CONFIG_SPL_DECOMP_HEADER is not set
 CONFIG_SPL_BOARD_INIT=y
+# CONFIG_SPL_BOOTROM_SUPPORT is not set
 # CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
 # CONFIG_SPL_LEGACY_IMAGE_SUPPORT is not set
+CONFIG_SPL_SYS_MALLOC_SIMPLE=y
+# CONFIG_TPL_SYS_MALLOC_SIMPLE is not set
+# CONFIG_SPL_STACK_R is not set
 CONFIG_SPL_SEPARATE_BSS=y
+# CONFIG_SPL_DISPLAY_PRINT is not set
+CONFIG_SPL_SKIP_RELOCATE=y
+# CONFIG_SPL_BOOT_IMAGE is not set
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x4000
 CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_PARTITION=y
 CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_PARTITION=0x1
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_PARTITION_NAME="uboot"
+# CONFIG_SPL_CRC32_SUPPORT is not set
+# CONFIG_SPL_MD5_SUPPORT is not set
+# CONFIG_SPL_SHA1_SUPPORT is not set
+# CONFIG_SPL_SHA256_SUPPORT is not set
+# CONFIG_SPL_FIT_IMAGE_TINY is not set
+# CONFIG_SPL_CPU_SUPPORT is not set
+# CONFIG_SPL_CRYPTO_SUPPORT is not set
+# CONFIG_SPL_HASH_SUPPORT is not set
+# CONFIG_SPL_DMA_SUPPORT is not set
+# CONFIG_SPL_ENV_SUPPORT is not set
+# CONFIG_SPL_EXT_SUPPORT is not set
+# CONFIG_SPL_FPGA_SUPPORT is not set
+# CONFIG_SPL_I2C_SUPPORT is not set
 CONFIG_SPL_MMC_WRITE=y
-CONFIG_SPL_MTD_SUPPORT=y
+# CONFIG_SPL_MPC8XXX_INIT_DDR_SUPPORT is not set
+# CONFIG_SPL_MTD_SUPPORT is not set
+# CONFIG_SPL_MUSB_NEW_SUPPORT is not set
+# CONFIG_SPL_NET_SUPPORT is not set
+# CONFIG_SPL_NO_CPU_SUPPORT is not set
+# CONFIG_SPL_NOR_SUPPORT is not set
+# CONFIG_SPL_XIP_SUPPORT is not set
+# CONFIG_SPL_ONENAND_SUPPORT is not set
+# CONFIG_SPL_OS_BOOT is not set
+# CONFIG_SPL_PCI_SUPPORT is not set
+# CONFIG_SPL_PCH_SUPPORT is not set
+# CONFIG_SPL_PCIE_EP_SUPPORT is not set
+# CONFIG_SPL_POST_MEM_SUPPORT is not set
+# CONFIG_SPL_POWER_SUPPORT is not set
+# CONFIG_SPL_PWM_SUPPORT is not set
+# CONFIG_SPL_RAM_SUPPORT is not set
+# CONFIG_SPL_RTC_SUPPORT is not set
+# CONFIG_SPL_SATA_SUPPORT is not set
+# CONFIG_SPL_RKNAND_SUPPORT is not set
+# CONFIG_SPL_SPI_FLASH_TINY is not set
+# CONFIG_SPL_SPI_FLASH_SFDP_SUPPORT is not set
+# CONFIG_SPL_SPI_LOAD is not set
+# CONFIG_SPL_USB_HOST_SUPPORT is not set
+# CONFIG_SPL_USB_GADGET is not set
+# CONFIG_SPL_YMODEM_SUPPORT is not set
 CONFIG_SPL_ATF=y
+# CONFIG_SPL_OPTEE_SUPPORT is not set
+# CONFIG_SPL_ATF_NO_PLATFORM_PARAM is not set
+# CONFIG_SPL_ATF_AARCH32_BL33 is not set
+# CONFIG_SPL_OPTEE is not set
 CONFIG_SPL_AB=y
+# CONFIG_SPL_LOAD_RKFW is not set
+# CONFIG_SPL_KERNEL_BOOT is not set
+CONFIG_TPL=y
+# CONFIG_TPL_BOARD_INIT is not set
+# CONFIG_TPL_NEEDS_SEPARATE_TEXT_BASE is not set
+# CONFIG_TPL_NEEDS_SEPARATE_STACK is not set
+# CONFIG_TPL_BOOTROM_SUPPORT is not set
+# CONFIG_TPL_DRIVERS_MISC_SUPPORT is not set
+# CONFIG_TPL_ENV_SUPPORT is not set
+# CONFIG_TPL_I2C_SUPPORT is not set
+CONFIG_TPL_TINY_FRAMEWORK=y
+# CONFIG_TPL_MPC8XXX_INIT_DDR_SUPPORT is not set
+# CONFIG_TPL_MMC_SUPPORT is not set
+# CONFIG_TPL_NAND_SUPPORT is not set
+CONFIG_TPL_SERIAL_SUPPORT=y
+# CONFIG_TPL_SPI_FLASH_SUPPORT is not set
+# CONFIG_TPL_SPI_SUPPORT is not set
+
+#
+# USBPLUG
+#
+# CONFIG_SUPPORT_USBPLUG is not set
+
+#
+# Command line interface
+#
+CONFIG_CMDLINE=y
+CONFIG_HUSH_PARSER=y
+CONFIG_SYS_PROMPT="=> "
+
+#
+# Autoboot options
+#
+CONFIG_AUTOBOOT=y
 CONFIG_AUTOBOOT_KEYED=y
 CONFIG_AUTOBOOT_PROMPT="Autoboot in %d seconds, press X to interrupt\n"
+# CONFIG_AUTOBOOT_ENCRYPTION is not set
+CONFIG_AUTOBOOT_DELAY_STR=""
 CONFIG_AUTOBOOT_STOP_STR="x"
+# CONFIG_AUTOBOOT_KEYED_CTRLC is not set
+
+#
+# FASTBOOT
+#
+CONFIG_FASTBOOT=y
+CONFIG_USB_FUNCTION_FASTBOOT=y
+# CONFIG_UDP_FUNCTION_FASTBOOT is not set
+CONFIG_CMD_FASTBOOT=y
 CONFIG_FASTBOOT_BUF_ADDR=0xc00800
 CONFIG_FASTBOOT_BUF_SIZE=0x07000000
+CONFIG_FASTBOOT_USB_DEV=0
 CONFIG_FASTBOOT_FLASH=y
 CONFIG_FASTBOOT_FLASH_MMC_DEV=0
-CONFIG_CMD_BOOTZ=y
+# CONFIG_FASTBOOT_OEM_UNLOCK is not set
+# CONFIG_RK_AVB_LIBAVB_ENABLE_ATH_UNLOCK is not set
+
+#
+# Commands
+#
+
+#
+# Info commands
+#
+# CONFIG_CMD_BDI is not set
+# CONFIG_CMD_CONFIG is not set
+# CONFIG_CMD_CONSOLE is not set
+# CONFIG_CMD_CPU is not set
+# CONFIG_CMD_LICENSE is not set
+
+#
+# Boot commands
+#
+CONFIG_CMD_BOOTD=y
+CONFIG_CMD_BOOTM=y
+# CONFIG_CMD_BOOTZ is not set
+CONFIG_CMD_BOOTI=y
+# CONFIG_CMD_BOOTMENU is not set
 CONFIG_CMD_DTIMG=y
 # CONFIG_CMD_ELF is not set
+# CONFIG_CMD_FDT is not set
+CONFIG_CMD_GO=y
+CONFIG_CMD_RUN=y
 # CONFIG_CMD_IMI is not set
 # CONFIG_CMD_IMLS is not set
 # CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_POWEROFF is not set
+# CONFIG_CMD_SPL is not set
+# CONFIG_CMD_THOR_DOWNLOAD is not set
+# CONFIG_CMD_ZBOOT is not set
+
+#
+# Environment commands
+#
+# CONFIG_CMD_ASKENV is not set
+# CONFIG_CMD_EXPORTENV is not set
+# CONFIG_CMD_IMPORTENV is not set
+CONFIG_CMD_EDITENV=y
+# CONFIG_CMD_GREPENV is not set
+CONFIG_CMD_SAVEENV=y
+CONFIG_CMD_ENV_EXISTS=y
+# CONFIG_CMD_ENV_CALLBACK is not set
+# CONFIG_CMD_ENV_FLAGS is not set
+
+#
+# Memory commands
+#
+CONFIG_CMD_CRC32=y
+# CONFIG_CRC32_VERIFY is not set
+# CONFIG_CMD_EEPROM is not set
+# CONFIG_LOOPW is not set
+# CONFIG_CMD_MD5SUM is not set
+# CONFIG_CMD_MEMINFO is not set
+# CONFIG_CMD_MEMORY is not set
+# CONFIG_CMD_MEMTEST is not set
+# CONFIG_CMD_MX_CYCLIC is not set
+# CONFIG_CMD_SHA1SUM is not set
+# CONFIG_CMD_STRINGS is not set
+# CONFIG_CMD_CRYPTO is not set
+# CONFIG_CMD_CRYPTO_SUM is not set
+
+#
+# Compression commands
+#
 # CONFIG_CMD_LZMADEC is not set
 # CONFIG_CMD_UNZIP is not set
+# CONFIG_CMD_ZIP is not set
+
+#
+# Device access commands
+#
+# CONFIG_CMD_ARMFLASH is not set
 CONFIG_CMD_CLK=y
+# CONFIG_CMD_DEMO is not set
+# CONFIG_CMD_DFU is not set
+CONFIG_CMD_DM=y
+# CONFIG_CMD_FDC is not set
 # CONFIG_CMD_FLASH is not set
 # CONFIG_CMD_FPGA is not set
+# CONFIG_CMD_FPGAD is not set
+# CONFIG_CMD_FUSE is not set
+# CONFIG_CMD_GPIO is not set
 CONFIG_CMD_GPT=y
+CONFIG_RANDOM_UUID=y
+# CONFIG_CMD_GPT_RENAME is not set
+# CONFIG_CMD_IDE is not set
+# CONFIG_CMD_IO is not set
+# CONFIG_CMD_IOTRACE is not set
+# CONFIG_CMD_I2C is not set
 # CONFIG_CMD_LOADB is not set
 # CONFIG_CMD_LOADS is not set
+# CONFIG_CMD_LOAD_ANDROID is not set
 CONFIG_CMD_BOOT_ANDROID=y
+CONFIG_CMD_BOOT_FIT=y
+# CONFIG_CMD_BOOT_ROCKCHIP is not set
 CONFIG_CMD_MMC=y
-CONFIG_CMD_SF=y
-CONFIG_CMD_SPI=y
+# CONFIG_CMD_MTD is not set
+# CONFIG_CMD_NAND is not set
+# CONFIG_CMD_MMC_SPI is not set
+# CONFIG_CMD_ONENAND is not set
+CONFIG_CMD_PART=y
+# CONFIG_CMD_PCI is not set
+# CONFIG_CMD_PCMCIA is not set
+# CONFIG_CMD_PINMUX is not set
+# CONFIG_CMD_READ is not set
+CONFIG_CMD_ROCKUSB=y
+# CONFIG_CMD_RK_SECURE_STORAGE is not set
+# CONFIG_CMD_SATA is not set
+# CONFIG_CMD_SAVES is not set
+# CONFIG_CMD_SDRAM is not set
+# CONFIG_CMD_SF is not set
+# CONFIG_CMD_SF_TEST is not set
+# CONFIG_CMD_SPI is not set
+# CONFIG_CMD_TSI148 is not set
+# CONFIG_CMD_UNIVERSE is not set
 CONFIG_CMD_USB=y
+# CONFIG_CMD_USB_SDP is not set
 CONFIG_CMD_USB_MASS_STORAGE=y
+
+#
+# Shell scripting commands
+#
+CONFIG_CMD_ECHO=y
 # CONFIG_CMD_ITEST is not set
+# CONFIG_CMD_SCRIPT_UPDATE is not set
+CONFIG_CMD_SOURCE=y
 # CONFIG_CMD_SETEXPR is not set
+
+#
+# Network commands
+#
 # CONFIG_CMD_NET is not set
+# CONFIG_CMD_TFTPPUT is not set
+# CONFIG_CMD_TFTP_BOOTM is not set
+# CONFIG_CMD_TFTP_FLASH is not set
+# CONFIG_CMD_TFTP_UPDATE is not set
+# CONFIG_CMD_TFTPSRV is not set
+# CONFIG_CMD_RARP is not set
+CONFIG_CMD_DHCP=y
 # CONFIG_CMD_PXE is not set
 # CONFIG_CMD_NFS is not set
 # CONFIG_CMD_MII is not set
+CONFIG_CMD_PING=y
+# CONFIG_CMD_CDP is not set
+# CONFIG_CMD_SNTP is not set
+# CONFIG_CMD_DNS is not set
+# CONFIG_CMD_LINK_LOCAL is not set
+# CONFIG_CMD_ETHSW is not set
+
+#
+# DDR Tool
+#
+# CONFIG_CMD_DDR_TOOL is not set
+
+#
+# Misc commands
+#
+# CONFIG_CMD_BMP is not set
+# CONFIG_CMD_BSP is not set
+# CONFIG_CMD_BKOPS_ENABLE is not set
+# CONFIG_CMD_CACHE is not set
+# CONFIG_CMD_DISPLAY is not set
+# CONFIG_CMD_LED is not set
+# CONFIG_CMD_DATE is not set
+# CONFIG_CMD_TIME is not set
+# CONFIG_CMD_GETTIME is not set
+# CONFIG_CMD_RNG is not set
 # CONFIG_CMD_MISC is not set
+# CONFIG_CMD_TIMER is not set
+# CONFIG_CMD_QFW is not set
+# CONFIG_CMD_TERMINAL is not set
+# CONFIG_CMD_UUID is not set
+
+#
+# Power commands
+#
+CONFIG_CMD_PMIC=y
 CONFIG_CMD_REGULATOR=y
+CONFIG_CMD_CHARGE_DISPLAY=y
+
+#
+# Security commands
+#
+# CONFIG_CMD_AES is not set
+# CONFIG_CMD_BLOB is not set
+# CONFIG_CMD_HASH is not set
+
+#
+# Firmware commands
+#
+
+#
+# Filesystem commands
+#
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+# CONFIG_CMD_EXT4_WRITE is not set
+# CONFIG_CMD_EXT4_SPARSE_WRITE is not set
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+# CONFIG_CMD_FS_UUID is not set
+# CONFIG_CMD_JFFS2 is not set
+# CONFIG_CMD_MTDPARTS is not set
 CONFIG_CMD_MTD_BLK=y
+# CONFIG_CMD_REISER is not set
+# CONFIG_CMD_SCSI is not set
+# CONFIG_CMD_ZFS is not set
+
+#
+# Debug commands
+#
+CONFIG_CMD_ATAGS=y
+# CONFIG_CMD_BEDBUG is not set
+# CONFIG_CMD_DIAG is not set
+# CONFIG_CMD_KGDB is not set
+# CONFIG_CMD_LOG is not set
+# CONFIG_CMD_TRACE is not set
+# CONFIG_CMD_UBI is not set
+
+#
+# Partition Types
+#
+CONFIG_PARTITIONS=y
+# CONFIG_MAC_PARTITION is not set
+# CONFIG_SPL_MAC_PARTITION is not set
+CONFIG_DOS_PARTITION=y
 # CONFIG_SPL_DOS_PARTITION is not set
 # CONFIG_ISO_PARTITION is not set
+# CONFIG_SPL_ISO_PARTITION is not set
+# CONFIG_AMIGA_PARTITION is not set
+# CONFIG_SPL_AMIGA_PARTITION is not set
+CONFIG_EFI_PARTITION=y
 CONFIG_EFI_PARTITION_ENTRIES_NUMBERS=64
+CONFIG_EFI_PARTITION_ENTRIES_OFF=0
+CONFIG_SPL_EFI_PARTITION=y
+CONFIG_PARTITION_UUIDS=y
+CONFIG_SPL_PARTITION_UUIDS=y
+# CONFIG_PARTITION_TYPE_GUID is not set
+# CONFIG_ENV_PARTITION is not set
+# CONFIG_SPL_ENV_PARTITION is not set
+# CONFIG_RKPARM_PARTITION is not set
+# CONFIG_SPL_ROCKCHIP_PARTITION is not set
+CONFIG_SUPPORT_OF_CONTROL=y
+CONFIG_DTC=y
+
+#
+# Device Tree Control
+#
+CONFIG_OF_CONTROL=y
+# CONFIG_OF_BOARD_FIXUP is not set
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_SPL_DTB_MINIMUM=y
+# CONFIG_TPL_OF_CONTROL is not set
 CONFIG_OF_LIVE=y
+CONFIG_OF_SEPARATE=y
+# CONFIG_OF_EMBED is not set
+# CONFIG_OF_BOARD is not set
+# CONFIG_OF_PRIOR_STAGE is not set
+CONFIG_OF_LIST="rk3588s-fydetab-duo"
+# CONFIG_MULTI_DTB_FIT is not set
+# CONFIG_SPL_MULTI_DTB_FIT is not set
 CONFIG_OF_SPL_REMOVE_PROPS="interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
+CONFIG_OF_U_BOOT_REMOVE_PROPS=""
+# CONFIG_SPL_OF_PLATDATA is not set
+CONFIG_MKIMAGE_DTC_PATH="dtc"
+
+#
+# Environment
+#
+CONFIG_ENV_IS_NOWHERE=y
+# CONFIG_ENV_IS_IN_EEPROM is not set
+# CONFIG_ENV_IS_IN_FAT is not set
+# CONFIG_ENV_IS_IN_FLASH is not set
+# CONFIG_ENV_IS_IN_MMC is not set
+# CONFIG_ENV_IS_IN_NAND is not set
+# CONFIG_ENV_IS_IN_NVRAM is not set
+# CONFIG_ENV_IS_IN_ONENAND is not set
+# CONFIG_ENV_IS_IN_REMOTE is not set
+# CONFIG_ENV_IS_IN_SPI_FLASH is not set
+# CONFIG_ENV_IS_IN_UBI is not set
+# CONFIG_ENV_IS_IN_BLK_DEV is not set
+# CONFIG_ENV_AES is not set
+# CONFIG_ENVF is not set
+# CONFIG_ENV_MEM_LAYOUT is not set
+# CONFIG_NET is not set
+
+#
+# Device Drivers
+#
+
+#
+# Generic Driver Options
+#
+CONFIG_DM=y
+CONFIG_SPL_DM=y
+# CONFIG_TPL_DM is not set
+CONFIG_DM_WARN=y
+# CONFIG_DM_DEBUG is not set
+CONFIG_DM_DEVICE_REMOVE=y
+CONFIG_DM_STDIO=y
+CONFIG_DM_SEQ_ALIAS=y
+# CONFIG_SPL_DM_SEQ_ALIAS is not set
 CONFIG_REGMAP=y
 CONFIG_SPL_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_SPL_SYSCON=y
+# CONFIG_DEVRES is not set
+CONFIG_SIMPLE_BUS=y
+CONFIG_SPL_SIMPLE_BUS=y
+CONFIG_OF_TRANSLATE=y
+# CONFIG_SPL_OF_TRANSLATE is not set
+CONFIG_ADC=y
+# CONFIG_ADC_REQ_REGULATOR is not set
+# CONFIG_ADC_EXYNOS is not set
+# CONFIG_ADC_SANDBOX is not set
 # CONFIG_SARADC_ROCKCHIP is not set
 CONFIG_SARADC_ROCKCHIP_V2=y
+# CONFIG_SATA is not set
+# CONFIG_SCSI_AHCI is not set
+
+#
+# SATA/SCSI device support
+#
+CONFIG_BLK=y
+CONFIG_SPL_BLK=y
+# CONFIG_SPL_BLK_READ_PREPARE is not set
+# CONFIG_BLOCK_CACHE is not set
+# CONFIG_IDE is not set
+
+#
+# Boot count support
+#
+# CONFIG_BOOTCOUNT is not set
+
+#
+# Clock
+#
 CONFIG_CLK=y
 CONFIG_SPL_CLK=y
 CONFIG_CLK_SCMI=y
 CONFIG_SPL_CLK_SCMI=y
+# CONFIG_CLK_AT91 is not set
+CONFIG_ROCKCHIP_CLK_PLL=y
+# CONFIG_CPU is not set
+# CONFIG_AMP is not set
+
+#
+# Hardware crypto devices
+#
 CONFIG_DM_CRYPTO=y
 CONFIG_SPL_DM_CRYPTO=y
+# CONFIG_FSL_CAAM is not set
+# CONFIG_SYS_FSL_SEC_BE is not set
+# CONFIG_SYS_FSL_SEC_LE is not set
+CONFIG_ROCKCHIP_RSA=y
+CONFIG_SPL_ROCKCHIP_RSA=y
+# CONFIG_ROCKCHIP_CIPHER is not set
+# CONFIG_SPL_ROCKCHIP_CIPHER is not set
+# CONFIG_ROCKCHIP_HMAC is not set
+# CONFIG_SPL_ROCKCHIP_HMAC is not set
+# CONFIG_ROCKCHIP_CRYPTO_V1 is not set
+# CONFIG_SPL_ROCKCHIP_CRYPTO_V1 is not set
 CONFIG_ROCKCHIP_CRYPTO_V2=y
 CONFIG_SPL_ROCKCHIP_CRYPTO_V2=y
 CONFIG_DM_RNG=y
 CONFIG_RNG_ROCKCHIP=y
+
+#
+# Demo for driver model
+#
+# CONFIG_DM_DEMO is not set
+
+#
+# DFU support
+#
+
+#
+# DMA Support
+#
+# CONFIG_DMA is not set
+# CONFIG_TI_EDMA3 is not set
+CONFIG_FIRMWARE=y
+CONFIG_ARM_PSCI_FW=y
 CONFIG_SCMI_FIRMWARE=y
 CONFIG_SPL_SCMI_FIRMWARE=y
+
+#
+# FPGA support
+#
+# CONFIG_FPGA_ALTERA is not set
+# CONFIG_FPGA_SOCFPGA is not set
+# CONFIG_FPGA_XILINX is not set
+# CONFIG_FPGA_ROCKCHIP is not set
+
+#
+# GPIO Support
+#
+CONFIG_DM_GPIO=y
 CONFIG_GPIO_HOG=y
+# CONFIG_GPIO_NO_UC_FLAG_SEQ_ALIAS is not set
+# CONFIG_ALTERA_PIO is not set
+# CONFIG_DWAPB_GPIO is not set
+# CONFIG_AT91_GPIO is not set
+# CONFIG_ATMEL_PIO4 is not set
+# CONFIG_INTEL_BROADWELL_GPIO is not set
+# CONFIG_INTEL_ICH6_GPIO is not set
+# CONFIG_IMX_RGPIO2P is not set
+# CONFIG_LPC32XX_GPIO is not set
+# CONFIG_MSM_GPIO is not set
+# CONFIG_CMD_PCA953X is not set
+# CONFIG_PCF8575_GPIO is not set
 CONFIG_ROCKCHIP_GPIO=y
 CONFIG_ROCKCHIP_GPIO_V2=y
+# CONFIG_CMD_TCA642X is not set
+# CONFIG_TEGRA_GPIO is not set
+# CONFIG_TEGRA186_GPIO is not set
+# CONFIG_VYBRID_GPIO is not set
+# CONFIG_DM_74X164 is not set
+# CONFIG_DM_PCA953X is not set
+# CONFIG_MPC85XX_GPIO is not set
+# CONFIG_NCA9539_GPIO is not set
+
+#
+# I2C support
+#
+CONFIG_DM_I2C=y
+# CONFIG_DM_I2C_COMPAT is not set
+# CONFIG_I2C_SET_DEFAULT_BUS_NUM is not set
+# CONFIG_DM_I2C_GPIO is not set
+# CONFIG_SYS_I2C_FSL is not set
+# CONFIG_SYS_I2C_CADENCE is not set
+# CONFIG_SYS_I2C_DW is not set
+# CONFIG_SYS_I2C_INTEL is not set
+# CONFIG_SYS_I2C_IMX_LPI2C is not set
 CONFIG_SYS_I2C_ROCKCHIP=y
+# CONFIG_SYS_I2C_MVTWSI is not set
 CONFIG_I2C_MUX=y
+# CONFIG_SPL_I2C_MUX is not set
+# CONFIG_I2C_ARB_GPIO_CHALLENGE is not set
+# CONFIG_I2C_MUX_PCA954x is not set
+# CONFIG_I2C_MUX_GPIO is not set
+CONFIG_I2C_MUX_MAX96745=y
+CONFIG_I2C_MUX_MAX96755F=y
+CONFIG_INPUT=y
+# CONFIG_SPL_INPUT is not set
+# CONFIG_DM_KEYBOARD is not set
 CONFIG_DM_KEY=y
+# CONFIG_DM_RC is not set
+# CONFIG_SPL_DM_KEYBOARD is not set
+# CONFIG_CROS_EC_KEYB is not set
 CONFIG_RK8XX_PWRKEY=y
 CONFIG_ADC_KEY=y
+# CONFIG_GPIO_KEY is not set
+# CONFIG_RK_KEY is not set
+CONFIG_PWRKEY_DNL_TRIGGER_NUM=0
+
+#
+# LED Support
+#
+CONFIG_LED=y
+# CONFIG_LED_BLINK is not set
+# CONFIG_SPL_LED is not set
+CONFIG_LED_GPIO=y
+# CONFIG_LED_STATUS is not set
+
+#
+# Mailbox Controller Support
+#
+# CONFIG_DM_MAILBOX is not set
+
+#
+# Memory Controller drivers
+#
+
+#
+# Multifunction device drivers
+#
 CONFIG_MISC=y
 CONFIG_SPL_MISC=y
 CONFIG_MISC_DECOMPRESS=y
 CONFIG_SPL_MISC_DECOMPRESS=y
+# CONFIG_ALTERA_SYSID is not set
+# CONFIG_ATSHA204A is not set
+# CONFIG_ROCKCHIP_EFUSE is not set
 CONFIG_ROCKCHIP_OTP=y
 CONFIG_ROCKCHIP_HW_DECOMPRESS=y
+# CONFIG_ROCKCHIP_PM_CONFIG is not set
 CONFIG_SPL_ROCKCHIP_HW_DECOMPRESS=y
 CONFIG_SPL_ROCKCHIP_SECURE_OTP=y
+# CONFIG_CROS_EC is not set
+# CONFIG_DS4510 is not set
+# CONFIG_FSL_SEC_MON is not set
+# CONFIG_MXC_OCOTP is not set
+# CONFIG_NUVOTON_NCT6102D is not set
+# CONFIG_PWRSEQ is not set
+# CONFIG_PCA9551_LED is not set
+# CONFIG_WINBOND_W83627 is not set
+# CONFIG_I2C_EEPROM is not set
+
+#
+# MMC Host controller Support
+#
+CONFIG_MMC=y
+CONFIG_MMC_WRITE=y
+CONFIG_DM_MMC=y
+CONFIG_SPL_DM_MMC=y
+# CONFIG_SPL_MMC_TINY is not set
+# CONFIG_MMC_SIMPLE is not set
+CONFIG_SUPPORT_EMMC_RPMB=y
 CONFIG_MMC_DW=y
+# CONFIG_MMC_DW_K3 is not set
 CONFIG_MMC_DW_ROCKCHIP=y
+# CONFIG_MMC_MXC is not set
+# CONFIG_MMC_PCI is not set
+# CONFIG_MMC_OMAP_HS is not set
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_SDMA=y
+# CONFIG_MMC_SDHCI_BCMSTB is not set
+# CONFIG_MMC_SDHCI_CADENCE is not set
+# CONFIG_MMC_SDHCI_KONA is not set
+# CONFIG_MMC_SDHCI_MSM is not set
 CONFIG_MMC_SDHCI_ROCKCHIP=y
-CONFIG_MTD=y
-CONFIG_MTD_BLK=y
-CONFIG_MTD_DEVICE=y
-CONFIG_NAND=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_SPI_FLASH=y
+# CONFIG_MMC_SDHCI_S5P is not set
+# CONFIG_MMC_SDHCI_SPEAR is not set
+# CONFIG_MMC_SDHCI_STI is not set
+# CONFIG_MMC_SDHCI_XENON is not set
+# CONFIG_MMC_SDHCI_TANGIER is not set
+# CONFIG_MMC_USE_PRE_CONFIG is not set
+
+#
+# MTD Support
+#
+# CONFIG_MTD is not set
+# CONFIG_MTD_NOR_FLASH is not set
+# CONFIG_MTD_DEVICE is not set
+# CONFIG_FLASH_CFI_DRIVER is not set
+# CONFIG_MTD_NAND_BBT_USING_FLASH is not set
+# CONFIG_NAND is not set
+
+#
+# SPI Flash Support
+#
+CONFIG_DM_SPI_FLASH=y
+# CONFIG_SPI_FLASH is not set
+CONFIG_SF_DEFAULT_BUS=0
+CONFIG_SF_DEFAULT_CS=0
+CONFIG_SF_DEFAULT_MODE=3
 CONFIG_SF_DEFAULT_SPEED=80000000
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SST=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_SPI_FLASH_XMC=y
-CONFIG_SPI_FLASH_XTX=y
-CONFIG_SPI_FLASH_MTD=y
-CONFIG_PHYLIB=y
-CONFIG_DM_ETH=y
-CONFIG_DM_ETH_PHY=y
+
+#
+# UBI support
+#
+# CONFIG_CONFIG_UBI_SILENCE_MSG is not set
+# CONFIG_MTD_UBI is not set
+# CONFIG_BITBANGMII is not set
+# CONFIG_MV88E6352_SWITCH is not set
+# CONFIG_PHYLIB is not set
+# CONFIG_DM_ETH is not set
+# CONFIG_DM_ETH_PHY is not set
+# CONFIG_PCI is not set
+
+#
+# PHY Subsystem
+#
+CONFIG_PHY=y
+# CONFIG_SPL_PHY is not set
+# CONFIG_NOP_PHY is not set
 CONFIG_PHY_ROCKCHIP_INNO_USB2=y
+# CONFIG_PHY_ROCKCHIP_NANENG_COMBOPHY is not set
+# CONFIG_PHY_ROCKCHIP_NANENG_USB2 is not set
+# CONFIG_PHY_ROCKCHIP_INNO_USB3 is not set
+# CONFIG_PHY_ROCKCHIP_NANENG_EDP is not set
 CONFIG_PHY_ROCKCHIP_SAMSUNG_HDPTX=y
+# CONFIG_PHY_ROCKCHIP_SNPS_PCIE3 is not set
 CONFIG_PHY_ROCKCHIP_USBDP=y
+# CONFIG_PHY_ROCKCHIP_TYPEC is not set
+# CONFIG_MVEBU_COMPHY_SUPPORT is not set
+
+#
+# Pin controllers
+#
 CONFIG_PINCTRL=y
+CONFIG_PINCTRL_FULL=y
+CONFIG_PINCTRL_GENERIC=y
+CONFIG_PINMUX=y
+# CONFIG_PINCONF is not set
+CONFIG_PINCONF_RECURSIVE=y
 CONFIG_SPL_PINCTRL=y
+CONFIG_SPL_PINCTRL_FULL=y
+CONFIG_SPL_PINCTRL_GENERIC=y
+CONFIG_SPL_PINMUX=y
+# CONFIG_SPL_PINCONF is not set
+CONFIG_SPL_PINCONF_RECURSIVE=y
+# CONFIG_PINCTRL_AT91 is not set
+# CONFIG_PINCTRL_AT91PIO4 is not set
+CONFIG_PINCTRL_MAX96745=y
+CONFIG_PINCTRL_MAX96755F=y
+CONFIG_PINCTRL_ROCKCHIP=y
+CONFIG_SPL_PINCTRL_ROCKCHIP=y
+# CONFIG_PINCTRL_SINGLE is not set
+# CONFIG_PINCTRL_STM32 is not set
+
+#
+# Power
+#
+
+#
+# Power Domain Support
+#
+# CONFIG_POWER_DOMAIN is not set
 CONFIG_DM_FUEL_GAUGE=y
-CONFIG_POWER_FG_CW201X=y
-CONFIG_POWER_FG_CW221X=y
+# CONFIG_SPL_DM_FUEL_GAUGE is not set
+# CONFIG_POWER_FG_CW201X is not set
+# CONFIG_POWER_FG_CW221X is not set
+
+#
+# IO Domain Support
+#
+# CONFIG_IO_DOMAIN is not set
 CONFIG_DM_PMIC=y
+# CONFIG_SPL_DM_PMIC is not set
+CONFIG_PMIC_CHILDREN=y
+CONFIG_SPL_PMIC_CHILDREN=y
+# CONFIG_PMIC_ACT8846 is not set
+# CONFIG_PMIC_AS3722 is not set
+# CONFIG_DM_PMIC_PFUZE100 is not set
+# CONFIG_DM_PMIC_MAX77686 is not set
+# CONFIG_DM_PMIC_MAX8998 is not set
+# CONFIG_PMIC_MAX8997 is not set
+# CONFIG_PMIC_PM8916 is not set
+# CONFIG_PMIC_RK8XX is not set
 CONFIG_PMIC_SPI_RK8XX=y
+# CONFIG_PMIC_S2MPS11 is not set
+# CONFIG_DM_PMIC_SANDBOX is not set
+# CONFIG_PMIC_S5M8767 is not set
+# CONFIG_PMIC_RN5T567 is not set
+# CONFIG_PMIC_TPS65090 is not set
+# CONFIG_PMIC_PALMAS is not set
+# CONFIG_PMIC_LP873X is not set
+# CONFIG_PMIC_LP87565 is not set
+# CONFIG_POWER_MC34VR500 is not set
 CONFIG_DM_POWER_DELIVERY=y
 CONFIG_TYPEC_TCPM=y
 CONFIG_TYPEC_TCPCI=y
 CONFIG_TYPEC_HUSB311=y
-CONFIG_TYPEC_FUSB302=y
+# CONFIG_TYPEC_FUSB302 is not set
+CONFIG_DM_REGULATOR=y
+# CONFIG_SPL_DM_REGULATOR is not set
+# CONFIG_REGULATOR_FAN53555 is not set
 CONFIG_REGULATOR_PWM=y
 CONFIG_DM_REGULATOR_FIXED=y
+# CONFIG_SPL_DM_REGULATOR_FIXED is not set
 CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_REGULATOR_RK860X=y
-CONFIG_REGULATOR_RK806=y
+# CONFIG_REGULATOR_RK8XX is not set
+# CONFIG_DM_DVFS is not set
 CONFIG_CHARGER_BQ25700=y
-CONFIG_CHARGER_BQ25890=y
-CONFIG_CHARGER_SC8551=y
-CONFIG_CHARGER_SGM41542=y
+# CONFIG_CHARGER_BQ25890 is not set
+# CONFIG_CHARGER_SC8551 is not set
+# CONFIG_CHARGER_SGM41542 is not set
 CONFIG_DM_CHARGE_DISPLAY=y
 CONFIG_CHARGE_ANIMATION=y
+CONFIG_ROCKCHIP_PM=y
+CONFIG_DM_PWM=y
+# CONFIG_PWM_EXYNOS is not set
 CONFIG_PWM_ROCKCHIP=y
+# CONFIG_PWM_SANDBOX is not set
+# CONFIG_PWM_TEGRA is not set
 CONFIG_RAM=y
 CONFIG_SPL_RAM=y
 CONFIG_TPL_RAM=y
-CONFIG_DM_RAMDISK=y
-CONFIG_RAMDISK_RO=y
+# CONFIG_STM32_SDRAM is not set
+# CONFIG_DM_RAMDISK is not set
+# CONFIG_DM_DMC is not set
+# CONFIG_ROCKCHIP_DMC is not set
+# CONFIG_ROCKCHIP_SDRAM_COMMON is not set
+# CONFIG_ROCKCHIP_DRAM_EXTENDED_TEMP_SUPPORT is not set
+CONFIG_ROCKCHIP_TPL_INIT_DRAM_TYPE=3
+
+#
+# Remote Processor drivers
+#
+
+#
+# Reset Controller Support
+#
 CONFIG_DM_RESET=y
 CONFIG_SPL_DM_RESET=y
+CONFIG_RESET_ROCKCHIP=y
 CONFIG_SPL_RESET_ROCKCHIP=y
+# CONFIG_RESET_SCMI is not set
+# CONFIG_RKNAND is not set
+# CONFIG_RKFLASH is not set
+
+#
+# Real Time Clock
+#
+# CONFIG_DM_RTC is not set
+# CONFIG_SCSI is not set
+# CONFIG_DM_SCSI is not set
+
+#
+# Serial drivers
+#
 CONFIG_BAUDRATE=1500000
+CONFIG_REQUIRE_SERIAL_CONSOLE=y
+CONFIG_SERIAL_PRESENT=y
+CONFIG_SPL_SERIAL_PRESENT=y
+CONFIG_DM_SERIAL=y
+# CONFIG_SERIAL_RX_BUFFER is not set
+CONFIG_SPL_DM_SERIAL=y
+CONFIG_TPL_DM_SERIAL=y
+# CONFIG_DEBUG_UART_ALTERA_JTAGUART is not set
+# CONFIG_DEBUG_UART_ALTERA_UART is not set
+# CONFIG_DEBUG_UART_ATMEL is not set
+CONFIG_DEBUG_UART_NS16550=y
+# CONFIG_DEBUG_UART_S5P is not set
+# CONFIG_DEBUG_UART_UARTLITE is not set
+# CONFIG_DEBUG_UART_ARM_DCC is not set
+# CONFIG_DEBUG_MVEBU_A3700_UART is not set
+# CONFIG_DEBUG_UART_ZYNQ is not set
+# CONFIG_DEBUG_UART_PL010 is not set
+# CONFIG_DEBUG_UART_PL011 is not set
+# CONFIG_DEBUG_UART_OMAP is not set
 CONFIG_DEBUG_UART_BASE=0xFEB50000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_DEBUG_UART_SHIFT=2
+CONFIG_DEBUG_UART_BOARD_INIT=y
+# CONFIG_DEBUG_UART_ANNOUNCE is not set
+# CONFIG_DEBUG_UART_SKIP_INIT is not set
+CONFIG_DEBUG_UART_ALWAYS=y
+# CONFIG_ALTERA_JTAG_UART is not set
+# CONFIG_ALTERA_UART is not set
+# CONFIG_ATMEL_USART is not set
+# CONFIG_FSL_LPUART is not set
+# CONFIG_MVEBU_A3700_UART is not set
+CONFIG_SYS_NS16550=y
+# CONFIG_MSM_SERIAL is not set
+# CONFIG_PXA_SERIAL is not set
+
+#
+# Sound support
+#
+# CONFIG_SOUND is not set
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+# CONFIG_SPI_MEM is not set
+# CONFIG_ALTERA_SPI is not set
+# CONFIG_ATCSPI200_SPI is not set
+# CONFIG_ATMEL_SPI is not set
+# CONFIG_BCMSTB_SPI is not set
+# CONFIG_CADENCE_QSPI is not set
+# CONFIG_DESIGNWARE_SPI is not set
+# CONFIG_EXYNOS_SPI is not set
+# CONFIG_FSL_DSPI is not set
+# CONFIG_ICH_SPI is not set
+# CONFIG_MVEBU_A3700_SPI is not set
 CONFIG_ROCKCHIP_SPI=y
-CONFIG_ROCKCHIP_SFC=y
+# CONFIG_ROCKCHIP_SFC is not set
+# CONFIG_TEGRA114_SPI is not set
+# CONFIG_TEGRA20_SFLASH is not set
+# CONFIG_TEGRA20_SLINK is not set
+# CONFIG_TEGRA210_QSPI is not set
+# CONFIG_XILINX_SPI is not set
+# CONFIG_SOFT_SPI is not set
+# CONFIG_CF_SPI is not set
+# CONFIG_FSL_ESPI is not set
+# CONFIG_FSL_QSPI is not set
+# CONFIG_SH_SPI is not set
+# CONFIG_SH_QSPI is not set
+# CONFIG_TI_QSPI is not set
+# CONFIG_KIRKWOOD_SPI is not set
+# CONFIG_LPC32XX_SSP is not set
+# CONFIG_MPC8XXX_SPI is not set
+# CONFIG_MXC_SPI is not set
+# CONFIG_MXS_SPI is not set
+# CONFIG_OMAP3_SPI is not set
+
+#
+# SPMI support
+#
+# CONFIG_SPMI is not set
+
+#
+# System reset device drivers
+#
 CONFIG_SYSRESET=y
+CONFIG_SPL_SYSRESET=y
+CONFIG_SYSRESET_PSCI=y
+# CONFIG_SYSRESET_SYSCON is not set
+CONFIG_SYSRESET_SYSCON_REBOOT=y
+# CONFIG_SYSRESET_WATCHDOG is not set
+# CONFIG_DM_THERMAL is not set
+
+#
+# Timer Support
+#
+# CONFIG_TIMER is not set
+
+#
+# TPM support
+#
 CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_SPL_DM_USB=y
+# CONFIG_DM_USB_GADGET is not set
+# CONFIG_SPL_DM_USB_GADGET is not set
+
+#
+# USB Host Controller Drivers
+#
+CONFIG_USB_HOST=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_XHCI_DWC3=y
+# CONFIG_USB_XHCI_DWC3_OF_SIMPLE is not set
+# CONFIG_USB_XHCI_PCI is not set
 CONFIG_USB_EHCI_HCD=y
+# CONFIG_USB_EHCI_MSM is not set
+# CONFIG_USB_EHCI_PCI is not set
 CONFIG_USB_EHCI_GENERIC=y
 CONFIG_USB_OHCI_HCD=y
+# CONFIG_USB_OHCI_PCI is not set
 CONFIG_USB_OHCI_GENERIC=y
+# CONFIG_USB_OHCI_DA8XX is not set
+# CONFIG_USB_UHCI_HCD is not set
+# CONFIG_USB_DWC2 is not set
+# CONFIG_USB_R8A66597_HCD is not set
+# CONFIG_USB_CDNS3 is not set
 CONFIG_USB_DWC3=y
+# CONFIG_USB_DWC3_HOST is not set
 CONFIG_USB_DWC3_GADGET=y
+
+#
+# Platform Glue Driver Support
+#
+# CONFIG_USB_DWC3_OMAP is not set
 CONFIG_USB_DWC3_GENERIC=y
+
+#
+# PHY Subsystem
+#
+# CONFIG_USB_DWC3_PHY_OMAP is not set
+# CONFIG_USB_DWC3_PHY_SAMSUNG is not set
+
+#
+# MUSB Controller Driver
+#
+# CONFIG_USB_MUSB_HOST is not set
+# CONFIG_USB_MUSB_GADGET is not set
+# CONFIG_USB_MUSB_DA8XX is not set
+# CONFIG_USB_MUSB_TI is not set
+
+#
+# ULPI drivers
+#
+
+#
+# USB peripherals
+#
 CONFIG_USB_STORAGE=y
+# CONFIG_USB_KEYBOARD is not set
 CONFIG_USB_GADGET=y
 CONFIG_USB_GADGET_MANUFACTURER="Rockchip"
 CONFIG_USB_GADGET_VENDOR_NUM=0x2207
 CONFIG_USB_GADGET_PRODUCT_NUM=0x350a
+# CONFIG_USB_GADGET_ATMEL_USBA is not set
+# CONFIG_USB_GADGET_BCM_UDC_OTG_PHY is not set
+# CONFIG_USB_GADGET_DWC2_OTG is not set
+# CONFIG_CI_UDC is not set
+CONFIG_USB_GADGET_VBUS_DRAW=2
+CONFIG_USB_GADGET_DUALSPEED=y
 CONFIG_USB_GADGET_DOWNLOAD=y
+# CONFIG_USB_FUNCTION_SDP is not set
+# CONFIG_USB_HOST_ETHER is not set
+
+#
+# UFS Host Controller Support
+#
+# CONFIG_TI_J721E_UFS is not set
+
+#
+# Graphics support
+#
 CONFIG_DM_VIDEO=y
+# CONFIG_SPL_DM_VIDEO is not set
+CONFIG_BACKLIGHT_PWM=y
+# CONFIG_BACKLIGHT_GPIO is not set
+CONFIG_VIDEO_BPP8=y
+CONFIG_VIDEO_BPP16=y
+CONFIG_VIDEO_BPP32=y
+CONFIG_CONSOLE_NORMAL=y
+# CONFIG_CONSOLE_ROTATION is not set
+# CONFIG_CONSOLE_TRUETYPE is not set
+CONFIG_SYS_WHITE_ON_BLACK=y
+
+#
+# TrueType Fonts
+#
+# CONFIG_VIDCONSOLE_AS_LCD is not set
+# CONFIG_VIDEO_VESA is not set
+# CONFIG_VIDEO_LCD_ANX9804 is not set
+# CONFIG_VIDEO_LCD_SSD2828 is not set
+# CONFIG_VIDEO_MVEBU is not set
+CONFIG_I2C_EDID=y
 CONFIG_DISPLAY=y
+# CONFIG_ATMEL_HLCD is not set
+# CONFIG_VIDEO_ROCKCHIP is not set
 CONFIG_DRM_ROCKCHIP=y
+CONFIG_DRM_MEM_RESERVED_SIZE_MBYTES=32
+CONFIG_DRM_DP_HELPER=y
 CONFIG_DRM_MAXIM_MAX96745=y
 CONFIG_DRM_MAXIM_MAX96755F=y
-CONFIG_DRM_PANEL_MAXIM_DESERIALIZER=y
+# CONFIG_DRM_PANEL_ROHM_BU18RL82 is not set
+# CONFIG_DRM_PANEL_MAXIM_MAX96752F is not set
+CONFIG_DRM_ROCKCHIP_PANEL=y
 CONFIG_DRM_ROHM_BU18XL82=y
+# CONFIG_DRM_ROCKCHIP_DW_HDMI is not set
 CONFIG_DRM_ROCKCHIP_DW_HDMI_QP=y
+# CONFIG_DRM_ROCKCHIP_INNO_HDMI is not set
+# CONFIG_ROCKCHIP_INNO_HDMI_PHY is not set
+# CONFIG_DRM_ROCKCHIP_INNO_MIPI_PHY is not set
+# CONFIG_DRM_ROCKCHIP_INNO_VIDEO_PHY is not set
+# CONFIG_DRM_ROCKCHIP_INNO_VIDEO_COMBO_PHY is not set
+CONFIG_DRM_MIPI_DSI=y
+CONFIG_DRM_DSC=y
+# CONFIG_DRM_ROCKCHIP_DW_MIPI_DSI is not set
 CONFIG_DRM_ROCKCHIP_DW_MIPI_DSI2=y
 CONFIG_DRM_ROCKCHIP_DW_DP=y
 CONFIG_DRM_ROCKCHIP_ANALOGIX_DP=y
+# CONFIG_DRM_ROCKCHIP_LVDS is not set
+# CONFIG_DRM_ROCKCHIP_RGB is not set
+# CONFIG_DRM_ROCKCHIP_RK618 is not set
+# CONFIG_DRM_ROCKCHIP_RK1000 is not set
 CONFIG_DRM_ROCKCHIP_SAMSUNG_MIPI_DCPHY=y
 CONFIG_PHY_ROCKCHIP_SAMSUNG_HDPTX_HDMI=y
+# CONFIG_DRM_ROCKCHIP_TVE is not set
+CONFIG_ROCKCHIP_CUBIC_LUT_SIZE=0
+
+#
+# driver for different display serdes
+#
+# CONFIG_SERDES_DISPLAY is not set
+# CONFIG_VIDEO_TEGRA20 is not set
+# CONFIG_VIDEO_TEGRA124 is not set
+CONFIG_VIDEO_BRIDGE=y
+# CONFIG_VIDEO_BRIDGE_PARADE_PS862X is not set
+# CONFIG_VIDEO_BRIDGE_NXP_PTN3460 is not set
+# CONFIG_VIDEO_BRIDGE_ANALOGIX_ANX6345 is not set
+CONFIG_CONSOLE_SCROLL_LINES=1
+# CONFIG_LCD is not set
+# CONFIG_ROCKCHIP_EINK is not set
+
+#
+# Watchdog Timer Support
+#
+# CONFIG_BCM2835_WDT is not set
+# CONFIG_ULP_WATCHDOG is not set
+# CONFIG_WDT is not set
+CONFIG_IRQ=y
+# CONFIG_PHYS_TO_BUS is not set
+
+#
+# File systems
+#
+# CONFIG_FS_CBFS is not set
+CONFIG_FS_FAT=y
+CONFIG_FAT_WRITE=y
+CONFIG_FS_FAT_MAX_CLUSTSIZE=65536
+# CONFIG_FS_JFFS2 is not set
+# CONFIG_FS_CRAMFS is not set
+# CONFIG_YAFFS2 is not set
+
+#
+# Library routines
+#
+# CONFIG_BCH is not set
+# CONFIG_CC_OPTIMIZE_LIBS_FOR_SPEED is not set
+CONFIG_SYS_HZ=1000
+CONFIG_SYS_STACK_SIZE=0x200000
 CONFIG_USE_TINY_PRINTF=y
+CONFIG_PANIC_HANG=y
 CONFIG_REGEX=y
 CONFIG_LIB_RAND=y
 CONFIG_SPL_TINY_MEMSET=y
+# CONFIG_TPL_TINY_MEMSET is not set
+CONFIG_SYSMEM=y
+CONFIG_BIDRAM=y
+# CONFIG_CMD_DHRYSTONE is not set
+
+#
+# Security support
+#
+# CONFIG_AES is not set
 CONFIG_RSA=y
 CONFIG_SPL_RSA=y
+CONFIG_RSA_SOFTWARE_EXP=y
 CONFIG_RSA_N_SIZE=0x200
 CONFIG_RSA_E_SIZE=0x10
 CONFIG_RSA_C_SIZE=0x20
+# CONFIG_TPM is not set
+
+#
+# Boot Configuration
+#
 CONFIG_XBC=y
+
+#
+# Hashing Support
+#
+# CONFIG_SHA1 is not set
+# CONFIG_SHA256 is not set
+# CONFIG_SHA512 is not set
+# CONFIG_SHA_HW_ACCEL is not set
+
+#
+# Compression Support
+#
 CONFIG_LZ4=y
 CONFIG_LZMA=y
+# CONFIG_SPL_LZMA is not set
+# CONFIG_LZO is not set
+# CONFIG_SPL_LZO is not set
+# CONFIG_SPL_GZIP is not set
 CONFIG_ERRNO_STR=y
+# CONFIG_HEXDUMP is not set
+CONFIG_OF_LIBFDT=y
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_SPL_OF_LIBFDT=y
+# CONFIG_TPL_OF_LIBFDT is not set
+# CONFIG_FDT_FIXUP_PARTITIONS is not set
+
+#
+# System tables
+#
+CONFIG_LIB_RATIONAL=y
+# CONFIG_SPL_LIB_RATIONAL is not set
 # CONFIG_EFI_LOADER is not set
 CONFIG_AVB_LIBAVB=y
+# CONFIG_SPL_AVB_LIBAVB is not set
+# CONFIG_TPL_AVB_LIBAVB is not set
 CONFIG_AVB_LIBAVB_AB=y
+# CONFIG_SPL_AVB_LIBAVB_AB is not set
+# CONFIG_TPL_AVB_LIBAVB_AB is not set
 CONFIG_AVB_LIBAVB_ATX=y
+# CONFIG_SPL_AVB_LIBAVB_ATX is not set
+# CONFIG_TPL_AVB_LIBAVB_ATX is not set
 CONFIG_AVB_LIBAVB_USER=y
+# CONFIG_AVB_VBMETA_PUBLIC_KEY_VALIDATE is not set
+# CONFIG_SPL_AVB_LIBAVB_USER is not set
+# CONFIG_TPL_AVB_LIBAVB_USER is not set
 CONFIG_RK_AVB_LIBAVB_USER=y
+# CONFIG_SPL_RK_AVB_LIBAVB_USER is not set
+# CONFIG_TPL_RK_AVB_LIBAVB_USER is not set
+# CONFIG_ROCKCHIP_PRELOADER_PUB_KEY is not set
 CONFIG_OPTEE_CLIENT=y
+# CONFIG_OPTEE_V1 is not set
 CONFIG_OPTEE_V2=y
 CONFIG_OPTEE_ALWAYS_USE_SECURITY_PARTITION=y
+CONFIG_OPTEE_SECURITY_LEVEL=0
+# CONFIG_UNIT_TEST is not set
 CONFIG_TEST_ROCKCHIP=y
+# CONFIG_LAVA_INFO is not set
diff --git a/drivers/video/drm/rockchip_display.c b/drivers/video/drm/rockchip_display.c
index 6c5b3adb9f..e2f66952d4 100644
--- a/drivers/video/drm/rockchip_display.c
+++ b/drivers/video/drm/rockchip_display.c
@@ -933,7 +933,6 @@ static int display_enable(struct display_state *state)
 	return 0;
 }
 
-/*
 static int display_disable(struct display_state *state)
 {
 	struct crtc_state *crtc_state = &state->crtc_state;
@@ -958,7 +957,6 @@ static int display_disable(struct display_state *state)
 
 	return 0;
 }
-*/
 
 static int display_check(struct display_state *state)
 {
@@ -1545,19 +1543,34 @@ void rockchip_show_fbbase(ulong fbbase)
 	}
 }
 
-int rockchip_show_bmp(const char *bmp)
+void rockchip_close_display(void)
 {
-	printf("Warning: logo %s is not shown due to display bug\n", bmp);
-	return 0;
+  struct display_state *s;
+  list_for_each_entry(s, &rockchip_display_list, head)
+    display_disable(s);
 }
 
-int rockchip_show_logo(void)
+int rockchip_show_bmp(const char *bmp)
 {
-	printf("Warning: uboot logo is not shown due to display bug\n");
-	return 0;
+	struct display_state *s;
+	int ret = 0;
+
+	if (!bmp) {
+    rockchip_close_display();
+		return -ENOENT;
+	}
+
+	list_for_each_entry(s, &rockchip_display_list, head) {
+		s->logo.mode = s->charge_logo_mode;
+		if (load_bmp_logo(&s->logo, bmp))
+			continue;
+		ret = display_logo(s);
+	}
+
+	return ret;
 }
 
-int rockchip_show_loader_logo(void)
+int rockchip_show_logo(void)
 {
 	struct display_state *s;
 	struct display_state *ms = NULL;
diff --git a/drivers/video/drm/rockchip_panel.c b/drivers/video/drm/rockchip_panel.c
index e26d6d7244..a29aee3bdd 100644
--- a/drivers/video/drm/rockchip_panel.c
+++ b/drivers/video/drm/rockchip_panel.c
@@ -316,12 +316,12 @@ static void panel_simple_prepare(struct rockchip_panel *panel)
 
 	if (dm_gpio_is_valid(&priv->avdd_gpio)) {
 		dm_gpio_set_value(&priv->avdd_gpio, 1);
-		mdelay(2);
+		mdelay(5);
 	}
 
 	if (dm_gpio_is_valid(&priv->avee_gpio)) {
 		dm_gpio_set_value(&priv->avee_gpio, 1);
-		mdelay(2);
+		mdelay(10);
 	}
 
 	if (dm_gpio_is_valid(&priv->reset_gpio))
@@ -330,9 +330,6 @@ static void panel_simple_prepare(struct rockchip_panel *panel)
 	if (plat->delay.reset)
 		mdelay(plat->delay.reset);
 
-	if (dm_gpio_is_valid(&priv->reset_gpio))
-		dm_gpio_set_value(&priv->reset_gpio, 0);
-
 	if (plat->delay.init)
 		mdelay(plat->delay.init);
 
@@ -374,18 +371,22 @@ static void panel_simple_unprepare(struct rockchip_panel *panel)
 		if (ret)
 			printf("failed to send off cmds: %d\n", ret);
 	}
-
-	if (dm_gpio_is_valid(&priv->reset_gpio))
-		dm_gpio_set_value(&priv->reset_gpio, 1);
-
+  mdelay(2);
 	if (dm_gpio_is_valid(&priv->enable_gpio))
 		dm_gpio_set_value(&priv->enable_gpio, 0);
+  mdelay(2);
 
-	if (dm_gpio_is_valid(&priv->avdd_gpio))
-		dm_gpio_set_value(&priv->avdd_gpio, 0);
+	if (dm_gpio_is_valid(&priv->reset_gpio))
+		dm_gpio_set_value(&priv->reset_gpio, 0);
+  mdelay(2);
 
 	if (dm_gpio_is_valid(&priv->avee_gpio))
 		dm_gpio_set_value(&priv->avee_gpio, 0);
+  mdelay(2);
+
+	if (dm_gpio_is_valid(&priv->avdd_gpio))
+		dm_gpio_set_value(&priv->avdd_gpio, 0);
+  mdelay(2);
 
 	if (priv->power_supply)
 		regulator_set_enable(priv->power_supply, plat->power_invert);
diff --git a/include/video_rockchip.h b/include/video_rockchip.h
index 02a6178ff6..f1ce5e6a52 100644
--- a/include/video_rockchip.h
+++ b/include/video_rockchip.h
@@ -24,5 +24,6 @@
 int rockchip_show_bmp(const char *bmp);
 int rockchip_show_logo(void);
 void rockchip_display_fixup(void *blob);
+void rockchip_close_display(void);
 
 #endif
diff --git a/make.sh b/make.sh
index 6051559124..5c5901f632 100755
--- a/make.sh
+++ b/make.sh
@@ -13,7 +13,7 @@ CMD_ARGS=$1
 ########################################### User can modify #############################################
 RKBIN_TOOLS=../rkbin/tools
 CROSS_COMPILE_ARM32=../prebuilts/gcc/linux-x86/arm/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-
-CROSS_COMPILE_ARM64=../prebuilts/gcc/linux-x86/aarch64/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+CROSS_COMPILE_ARM64=../prebuilts/gcc/linux-x86/aarch64/gcc-arm-10.3-2021.07-x86_64-aarch64-none-linux-gnu/bin/aarch64-none-linux-gnu-
 ########################################### User not touch #############################################
 # Declare global INI file searching index name for every chip, update in select_chip_info()
 RKCHIP=
@@ -277,7 +277,7 @@ function select_toolchain()
 			CROSS_COMPILE_ARM64=`cat ${CC_FILE}`
 		else
 			if grep -q '^CONFIG_ARM64=y' .config ; then
-				CROSS_COMPILE_ARM64=$(cd `dirname ${CROSS_COMPILE_ARM64}`; pwd)"/aarch64-linux-gnu-"
+				CROSS_COMPILE_ARM64=$(cd `dirname ${CROSS_COMPILE_ARM64}`; pwd)"/aarch64-none-linux-gnu-"
 			else
 				CROSS_COMPILE_ARM32=$(cd `dirname ${CROSS_COMPILE_ARM32}`; pwd)"/arm-linux-gnueabihf-"
 			fi
diff --git a/tools/rockchip/bmp2gray16.c b/tools/rockchip/bmp2gray16.c
index a331d14397..c533430ad1 100644
--- a/tools/rockchip/bmp2gray16.c
+++ b/tools/rockchip/bmp2gray16.c
@@ -106,7 +106,7 @@ struct input_img_info {
 struct input_img_info in_img_info[16];
 uint32_t screen_w;
 uint32_t screen_h;
-static const char version[4] = "1.00";
+static const char *version = "1.00";
 static const char *PROG;
 
 static const char *fix_path(const char *path)
